src/app/myworkflows/myworkflows.service.ts

```diff
@@ -14,39 +14,117 @@
     protected usersService: UsersService,
     protected workflowService: WorkflowService,
     protected workflowsService: WorkflowsService,
-    private matSnackBar: MatSnackBar
+    private myBioWorkflowsService: MyBioWorkflowsService,
+    private myServicesService: MyServicesService,
+    public matDialog: MatDialog
   ) {
     super();
   }
 
+  getMyEntries(userId: number, entryType: EntryType) {
+    if (entryType === EntryType.BioWorkflow) {
+      this.myBioWorkflowsService.getMyBioWorkflows(userId);
+    } else {
+      this.myServicesService.getMyServices(userId);
+    }
+  }
+
   getGroupIndex(groupEntries: any[], group: string): number {
     return groupEntries.findIndex(orgWorkflow => orgWorkflow.sourceControl + '/' + orgWorkflow.organization === group);
   }
 
+  public findEntryFromPath(path: string, orgWorkflows: Array<OrgWorkflowObject>): ExtendedWorkflow {
+    let matchingWorkflow: ExtendedWorkflow;
+    for (let i = 0; i < orgWorkflows.length; i++) {
+      matchingWorkflow = orgWorkflows[i].published.find((workflow: ExtendedWorkflow) => workflow.full_workflow_path === path);
+      if (matchingWorkflow) {
+        return matchingWorkflow;
+      }
+      matchingWorkflow = orgWorkflows[i].unpublished.find((workflow: ExtendedWorkflow) => workflow.full_workflow_path === path);
+      if (matchingWorkflow) {
+        return matchingWorkflow;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Grabs the workflow from the webservice and loads it
+   * @param workflow Selected workflow
+   */
+  selectEntry(workflow: ExtendedWorkflow | null, entryType: EntryType | null): void {
+    if (workflow && entryType) {
+      if (entryType === EntryType.BioWorkflow) {
+        this.myBioWorkflowsService.selectEntry(workflow.id, includesValidation);
+      } else {
+        this.myServicesService.selectEntry(workflow.id, includesValidation);
+      }
+    }
+  }
+
+  public getFirstPublishedEntry(orgWorkflows: Array<OrgWorkflowObject>): Workflow {
+    for (let i = 0; i < orgWorkflows.length; i++) {
+      const foundWorkflow = orgWorkflows[i]['entries'].find((workflow: Workflow) => {
+        return workflow.is_published === true;
+      });
+      if (foundWorkflow) {
+        return foundWorkflow;
+      }
+    }
+    return null;
+  }
+
+  public convertOldNamespaceObjectToOrgEntriesObject(nsWorkflows: Array<any>): Array<OrgWorkflowObject> {
+    const groupEntriesObject: Array<OrgWorkflowObject> = [];
+    for (let i = 0; i < nsWorkflows.length; i++) {
+      const orgWorkflowObject: OrgWorkflowObject = {
+        sourceControl: '',
+        organization: '',
+        published: [],
+        unpublished: []
+      };
+      const nsWorkflow: Array<Workflow> = nsWorkflows[i].entries;
+      orgWorkflowObject.sourceControl = nsWorkflows[i].sourceControl;
+      orgWorkflowObject.organization = nsWorkflows[i].organization;
+      orgWorkflowObject.published = nsWorkflow.filter((workflow: Workflow) => {
```

Comment:
This and the block on line 116 can be written without curly braces and an implicit return. Sort of torn on it myself; I like the ability to set a breakpoint more cleanly this way, but OTOH, my IDE recommends the no block statement when possible. Your call.